test/test_numpy.py

```diff
@@ -1,296 +0,0 @@
-# Copyright Contributors to the Pyro project.
-# SPDX-License-Identifier: Apache-2.0
-
-from collections import OrderedDict
-
-import numpy as np
-import pytest
-
-import funsor
-from funsor import Number, Variable, bint, reals
-from funsor.domains import Domain
-from funsor.interpreter import _USE_TCO, interpretation
-from funsor.numpy import Array, align_arrays
-from funsor.terms import lazy
-from funsor.testing import assert_equiv, check_funsor, random_array
-
-# FIXME rewrite stack-based interpreter to be compatible with unhashable data
-xfail_with_tco = pytest.mark.xfail(
```

Comment:
@eb8680 It seems that NumPy tests passed with the Tensor class. Do I need to pay attention to this mark?

Reply:
I believe the failures here were caused by the fact that `numpy.ndarray`s were unhashable and hence couldn't be used as dictionary keys.  I'm surprised it works now, I'd have to dig into why.

Reply:
Oops, it doesn't work...

Reply:
In that case you can just leave the xfail, it's not an important issue.